src/Swift.Bindings/src/Emitter/StringEmitter/Handler/MethodHandler.cs

```diff
@@ -659,29 +665,58 @@ private void EmitAsync(CSharpWriter csWriter, SwiftWriter swiftWriter)
             GCHandle handle = GCHandle.Alloc(task, GCHandleType.Normal);
             """);
 
+            int genericIndex = 0;
             string parameters = string.Join(
                 ", ",
                 new[]
                 {
-                    $"callback: @escaping ({(isEmptyTuple ? "" : $"{_env.MethodDecl.CSSignature.First().SwiftTypeSpec}, ")}Int64) -> Void",
+                    $"callback: @escaping ({(isEmptyTuple ? "" : $"{(_env.MethodDecl.CSSignature.First().IsGeneric ? _env.MethodDecl.GenericParameters[0].SugaredTypeName : _env.MethodDecl.CSSignature.First().SwiftTypeSpec)}, ")}Int64) -> Void",
                     "task: Int64"
                 }.Concat(
                     _env.MethodDecl.CSSignature
                         .Skip(1)
-                        .Select(p => $"{p.Name}: {p.SwiftTypeSpec}")
+                        .Select(p => $"{p.Name}: {(p.IsGeneric ? _env.MethodDecl.GenericParameters[genericIndex++].SugaredTypeName : p.SwiftTypeSpec)}")
```

Comment:
NIT: this becomes rather hard to understand - incrementing an indexing within a select within a concat.

Reply:
Yes, it is incremented only if a generic parameter is encountered. improved it with `.Find` expression.